### ISSUE TITLE
icons with more than one dot in file name are not displayed in task bar

### ISSUE THREAD

**Issue by MartinP1** [linuxmint/cinnamon-desktop#242]
Affected Version: Cinnamon 6.0.4.

Linux Mint 21.3 "Virginia" actual level

Method to reproduce:

- rename an icon file e.g. from "/usr/share/icons/hicolor/scalable/apps/Icon.svg" to "/usr/share/icons/hicolor/scalable/apps/Icon.bla.svg"
- configure the app by start menu to use the icon with new name

Effect:
- App link is shown in start menu with correct icon "Icon.bla.svg" (correct)
- Started App is displayed in task-bar with system(?) icon "gear" (INCORRECT)

Problem appars in Package "Frescobaldi" (Music sheet editor) of linux mint, that provides an icon named  "org.frescobaldi.Frescobaldi.svg
"

**Comment by fredcw** [comment]
The problem here is not the number of dots in the name as many icon names have more than one. I think the problem is  the capital "F" which is not allowed according to Freedesktop's Icon Naming Specification:

>As a further restriction, all icon names may only contain lowercase letters, numbers, underscore, dash, or period characters.

Best to report bug to app developer I think.

**Comment by MartinP1** [comment]
Thanks for your Reaction 

https://github.com/frescobaldi/frescobaldi/issues/1769

I am not very hopeful concerning a reaction...

**Comment by fedelibre** [comment]
Hi all

I'm in charge of Frescobaldi freedesktop stuff.

I'm not sure that the problem is really the capital F. I see what [the spec](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html) (very old, by the way, latest change in 2006?!?) says, but I also see that all the GNOME applications use the app name capitalized. Does the problem occur with these apps? If not, then the problem might be somewhere else.

**Comment by fedelibre** [comment]
@MartinP1 The first thing I would check is the presence of other Frescobaldi icons. You may have done manual installations and those icons might interfere...

Please paste the outoput of this command:

    sudo find / -name *rescobaldi.svg

**Comment by fedelibre** [comment]
In [the desktop file spec](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id) we see instructions which reflect how names are used in every application:

> The name of the desktop entry should follow the "reverse DNS" convention: it should start with a reversed DNS domain name controlled by the author of the application, in lower case. The domain name should be followed by the name of the application, which is conventionally written with words run together and initial capital letters (CamelCase). For example, if the owner of example.org writes "Foo Viewer", they might choose the name org.example.FooViewer, resulting in a file named org.example.FooViewer.desktop. 

I don't see why icon names should be written differently from the desktop file.

**Comment by fedelibre** [comment]
On the other hand, if I look at the [AppStream spec](https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic) I read something differently:

> Additionally, even though uppercase letters are permitted in a component-ID, it is strongly encouraged to only use lowercase letters for the ID.
>
>The ID must follow a reverse-DNS scheme, consisting of {tld}.{vendor}.{product}, for example org.kde.gwenview or com.hugski.colorhug2.

It might be a new direction that apps haven't picked up yet.

**Comment by fredcw** [comment]
@fedelibre You're right, I don't think it's a problem with the icon or the icon name. I think it's something to do with setting an icon for the application or the application's main window in the application itself, rather than the icon in the .desktop file. The problem also occurs in MATE desktop where a generic application icon is used in the taskbar as well. I haven't tested xfce.
